lib/galaxy/auth/providers/ldap_ad.py

```diff
@@ -282,11 +295,179 @@ def authenticate_user(self, user, password, options, request):
         return self.authenticate(user.email, user.username, password, options, request)[0]
 
 
+class LDAP3(LDAP):
+    """LDAP auth provider using ldap3 module"""
+
+    plugin_type = "ldap3"
+
+    def __init__(self):
+        super().__init__()
+        # Initialise server and autobind objects
+        self.server = None
+        self.auto_bind = None
+        # LDAP over TLS bool
+        self.ldap_tls = None
+
+    def get_server(self, options, params):
+        # Get server URL
+        server_url = _get_subs(options, "server", params)
+        # Check if server URL has scheme
+        # If no scheme is provided, assume it to be ldap
+        if "ldap" not in server_url:
+            server_url = "ldap://{server_url}".format(server_url=server_url)
+        # Check if TLS is available
+        if server_url.startswith("ldaps://"):
+            self.ldap_tls = True
+        else:
+            self.ldap_tls = False
+        try:
+            # Get server address and port
+            server_address, server_port = server_url.split("//")[-1].split(":")
```

Comment:
Is there something in the python stdlib to split URIs?

Reply:
Not that I know. For HTTP, there is one. Not sure if it works for LDAP. Will check it. Cheers